ci-operator/step-registry/ingress/aws/conf/subnets/ingress-aws-conf-subnets-commands.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+# For disconnected or otherwise unreachable environments, we want to
+# have steps use an HTTP(S) proxy to reach the API server. This proxy
+# configuration file should export HTTP_PROXY, HTTPS_PROXY, and NO_PROXY
+# environment variables, as well as their lowercase equivalents (note
+# that libcurl doesn't recognize the uppercase variables).
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1090
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+# output example:
+# public_subnets: "ci-op-0r5slpp6-0dd5e-8r4pp-subnet-public-us-east-1a","ci-op-0r5slpp6-0dd5e-8r4pp-subnet-public-us-east-1d"
+
+public_subnets=$(oc -n openshift-machine-api get machinesets.machine.openshift.io -ojsonpath='{.items[*].spec.template.spec.providerSpec.value.subnet.filters[].values}' | tr -d "[]" | sed 's/ /,/g' | sed 's/subnet-private/subnet-public/g')
```

Comment:
The subnet gotten from machine-api is private subnet or public subnet in the real world ?

Reply:
It is private subnet in `machinesets.machine` AFAIK, not sure if we can get public subnet directly from other resources?

Reply:
So here we are use private subnents to configure ingress as public subnet, that sounds some potential issues for me.

Reply:
Using `sed 's/subnet-private/subnet-public/g'` to replace `private` with `public`  in this command, any other suggestion?

Reply:
So here actually we are expecting to use the real **public** subnet to configure ingress, right?

Reply:
Yes, we use external LB in the two workflows so `public` subnets should be used.

Reply:
@yunjiang29 do you have idea to get the public subnet ?

@lihongan btw, if follow the current logic, use the private cluster to configure ingress, the provisioned LB will get a internal IP, but not a external IP, right?

Reply:
No, External/Internal is managed by ingresscontroller `spec...loadBalancer.scope`, if attaching private subnets to external LB, the resolved IP address (AWS LB service uses alias instead of IP) is still public IP but LB is unreachable.

Reply:
@lihongan 
1. will this step be used in byo-vpc workflow?
2. I believe this step is only applicable for 4.17 and above, correct?

Reply:
@yunjiang29 

1. No plan to use the step in byo-vpc workflow, currently the step is just used in the existing `ingress-lbtype-xxx` workflows. In addition we still have e2e tests to cover subnets feature on other aws workflows.
2. Yes, ingresscontoller supports to choose subents is applicable for 4.17 and above.

Reply:
> sed 's/subnet-private/subnet-public/g'
No issues if this is for 4.17+ installer created VPC.  I think we can add a comment.

Reply:
@lihongan do we have any check-point to check if this configuration works as expected? Is [1] one of them 


[1] https://github.com/openshift/release/pull/55505/files#diff-79fe9d369b2246746d21da1bc0bedd84f99e572676ed6dd9d2df927a5ab4ff89R38

Reply:
Yes, after patching the ingresscontroller/default, the LB service should be deleted manually then wait new one with subnets annotation to be provisioned, if that works then co/ingress, authentication, console should not be degraded.

Reply:
added comment of how to get the public subnets name.